tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.h

```diff
@@ -85,7 +85,7 @@ class GpuCudaMallocAsyncAllocator : public Allocator {
 
   void SetStream(void* stream) override {
 #if TF_CUDA_MALLOC_ASYNC_SUPPORTED
-    cuda_stream_ = reinterpret_cast<CUstream>(stream);
+    cuda_stream_ = *(reinterpret_cast<CUstream*>(stream));
```

Comment:
I did not notice this earlier, but a setter for an immutable property seems like a code smell to me.  Have you tried passing in the stream via the constructor somehow?

(Does not need to be addressed in this PR.)

Reply:
Mostly, the GpuDevice need the gpu_allocator. But the gpu_allocator need the stream created by the GpuDevice.
Fixing this would request a refactoring that will break existing API and I wasn't sure if those API was part of the public API or not. As I wanted this to get in TF2.5, I prefered to stay away from this refactoring.
Relevant code:
https://github.com/tensorflow/tensorflow/blob/aa08697dcb98135ee39ba00ee08b5c1a28cfde61/tensorflow/core/common_runtime/gpu/gpu_device.cc#L1414-L1422

(I also updated the PR description)
(I amended the test commit to remove a commented left over line)